spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/KubernetesConfigDataLocationResolver.java

```diff
@@ -188,12 +196,21 @@ protected PropertyHolder loadProperties(ConfigDataLocationResolverContext contex
 		}
 		kubernetesClientProperties.setNamespace(namespace);
 
-		ConfigMapConfigProperties configMapConfigProperties = binder
-				.bind(ConfigMapConfigProperties.PREFIX, ConfigMapConfigProperties.class)
-				.orElseGet(ConfigMapConfigProperties::new);
-		SecretsConfigProperties secretsProperties = binder
-				.bind(SecretsConfigProperties.PREFIX, SecretsConfigProperties.class)
-				.orElseGet(SecretsConfigProperties::new);
+		boolean configEnabled = binder.bind("spring.cloud.kubernetes.config.enabled", boolean.class).orElse(true);
```

Comment:
this is the idea of this PR. I noticed this small thing while working on a different thing, really. 

Instead of providing the configmap and secrets properties all the time (even if configmap/secret could be disabled), we can conditionally provide them. If you think about it, if you have set : `spring.cloud.kubernetes.secrets.enabled=false`, then you obviously don't care about `ConfigMapConfigProperties`...